src/main/resources/application.yml

```diff
@@ -1,11 +1,11 @@
 server:
-  port: 9999
+  port: ${SERVER_PORT:9999}
 
 spring:
   datasource:
-    url: jdbc:mysql://localhost:3306/hacktheinterview
-    username: root
-    password: password
+    url: ${DB_HOST:jdbc:mysql://localhost}:${DB_PORT:3306}/${DB_NAME:hacktheinterview}
```

Comment:
DB_HOST is only `localhost` not `jdbc:mysql://localhost`
Suggestion: `url: jdbc:mysql://${DB_HOST:localhost}`